src/RepairManager/rules/ecc_reboot_node_rule.py

```diff
@@ -43,131 +45,150 @@ def _create_email_for_pause_resume_job(job_id, node_names, job_link, job_owner_e
 
     def load_ecc_config(self):
         with open('/etc/RepairManager/config/ecc-config.yaml', 'r') as file:
             return yaml.safe_load(file)
 
+    def load_etcd_config(self):
+        with open('/etc/RepairManager/config/etcd.conf.yaml', 'r') as file:
+            return yaml.safe_load(file)
 
-    def check_status(self):
+    def check_if_nodes_have_rebooted(self):
+        # if node has been rebooted since ecc error initially detected,
+        # uncordon, remove from rule_cache, and mark as resolved
         url = f"http://{self.config['prometheus']['ip']}:{self.config['prometheus']['port']}"
         query = self.config['prometheus']['node_boot_time_query']
-        reboot_url = prometheus_util.format_url_query(url, query)
+        reboot_times_url = prometheus_util.format_url_query(url, query)
+        dry_run = not self.ecc_config["enable_reboot"]
+        uncordon_action = UncordonAction()
 
         try:
-            response = requests.get(reboot_url, timeout=10)
+            response = requests.get(reboot_times_url, timeout=10)
             if response:
                 reboot_data = response.json()
                 reboot_times = _extract_node_boot_time_info(reboot_data)
-                
-                # if node has been rebooted since ecc error first detected,
-                # remove from rule_cache
-                if self.rule in self.alert.rule_cache:
-                    remove_from_cache = []
-                    for node in self.alert.rule_cache[self.rule]:
-                        instance = self.alert.rule_cache[self.rule][node]["instance"]
-                        time_found_string = self.alert.rule_cache[self.rule][node]["time_found"]
-                        time_found_datetime = datetime.strptime(time_found_string, self.config['date_time_format'])
-                        last_reboot_time = reboot_times[instance]
-                        if last_reboot_time > time_found_datetime:
-                            remove_from_cache.append(node)
-                            activity_log.info({"action":"node marked as resolved","node":node})
-
-                    for node in remove_from_cache:
-                        self.alert.remove_from_rule_cache(self.rule, node)
+            bad_nodes = self.alert.get_rule_cache_keys(self.rule)
+            for node in bad_nodes:
+                instance = self.alert.get_rule_cache(self.rule, node)["instance"]
+                time_found_string = self.alert.get_rule_cache(self.rule, node)["time_found"]
+                time_found_datetime = datetime.strptime(time_found_string, self.config['date_time_format'])
+                last_reboot_time = reboot_times[instance]
+                if last_reboot_time > time_found_datetime:
+                    uncordon_action.execute(node, dry_run)
+                    self.alert.remove_from_rule_cache(self.rule, node)
+                    activity_log.info({"action":"marked as resolved from incorrectable ecc error","node":node})
```

Comment:
if rebooted since initial detection, uncordon node and mark as resolved by removing from the rule cache.